src/librustc_mir/borrow_check/diagnostics/move_errors.rs

```diff
@@ -383,10 +384,20 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                 }
             }
         };
-        let move_ty = format!("{:?}", move_place.ty(*self.body, self.infcx.tcx).ty,);
         if let Ok(snippet) = self.infcx.tcx.sess.source_map().span_to_snippet(span) {
-            let is_option = move_ty.starts_with("std::option::Option");
-            let is_result = move_ty.starts_with("std::result::Result");
+            let def_id = match move_place.ty(*self.body, self.infcx.tcx).ty.kind {
+                ty::Adt(self_def, _) => self_def.did,
```

Comment:
Result and Option can't be anything but `ty::Adt`, so you can use [`ty_adt_def`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyS.html#method.ty_adt_def) instead of manually matching on the `TyKind` variants